### ISSUE TITLE
can't launch games after prefix update

### ISSUE THREAD

**Issue by soredake** [lutris/lutris#2414]
**Describe the bug**
Lutris can't launch games if `drive_c/windows/system32/d3d9-nine.dll` is present in prefix.

**Expected behavior**

Launch a game.

**Current behavior**

What is the current behavior?

**Steps to reproduce**

- Update prefix with `WINEPREFIX=<prefix path> wineboot -u`
- Try to launch game

**Lutris debugging output**

The terminal output of `lutris -d`.

[lutris.log](https://github.com/lutris/lutris/files/3746846/lutris.log)

**System information**

The .json file created by typing `lutris --submit-issue` in a terminal.

**Comment by soredake** [comment]
@sonicpp

**Comment by sonicpp** [comment]
Yay, another issue with Nine? I gave it a quick look, but can not reproduce it yet. In the meantime, could you please check if you

- have `C:\Windows\System32\d3d9.dll` as a symlink to `C:\Windows\System32\d3d9-nine.dll`
- have _d3d9_ dll override set to _native_ (check _winecfg_ or _wine regedit_ (HKEY_CURRENT_USER : Software\Wine\DllOverrides)
- are running prefix with Nine compatible drivers

PS: I did not forget about your issue with nine-git package, dont worry

**Comment by soredake** [comment]
Prefix update creates `drive_c/windows/system32/d3d9-nine.dll`, if i delete these file games launch fine until next update. Gallium nine option is not enabled.

- d3d9.dll is symlink to d9vk.
- no d3d9 override.
- gallium drivers are installed.

log
```
Running /usr/bin/wine /home/bausch/.local/share/lutris/runners/winesteam/prefix64/drive_c/Program Files (x86)/Steam/Steam.exe -no-cef-sandbox -console steam://rungameid/648390
gamemodeauto: 
002c:err:fsync:fsync_init Failed to open fsync shared memory file; make sure no stale wineserver instances are running without WINEFSYNC.
Waiting on children
Exit with returncode 1
```

**Comment by sonicpp** [comment]
Thanks, Did you enable D9VK for your prefix before (you say d3d9 is linking to d9vk)? That might be a problem, since these two features (Nine and D9VK) where implemented in Lutris independently IMHO (another thing I want to take a look)

**Comment by soredake** [comment]
Yes, d9vk option is enabled in lutris, disabling it does not help.

**Comment by sonicpp** [comment]
Hi, I found the issue. Both technologies were integrated independently into Lutris, so there are some conflicts. According to your log, D9VK is enabled (sets D3D9 override to native - it expects existing d3d9.dll file), then Nine is disabled (it deletes d3d9.dll - this is a problem for D9VK).

But this should happen only if you enabled Nine before (you have d3d9-nine.dll in your folder) and you disabled it later and enabled D9VK instead. Disabling both Nine and D9VK, enabling both or enabling Nine while disabling D9VK should not cause a problem. Hmm

I am working on it.

**Comment by soredake** [comment]
I did not enabled nine before, `d3d9-nine.dll` added by prefix update.

**Comment by sonicpp** [comment]
Hmm... Odd. d3d9-nine.dll is copyied into system32 (or syswow64) only when you launch game with enabled Gallium Nine.

**Comment by Zeioth** [comment]
From D9VK release notes:

> Please do not use wine's d3d9x or d3dcompiler with d9vk. They make invalid API calls and generate bad shaders. I cannot stress this enough.

Try to delete create a clean wineprefix, and enable D9VK using the checkbox. Should work out of the box. If not, [make sure vulkan is correctly installed](https://github.com/lutris/lutris/wiki/How-to:-DXVK).
![dxvk](https://user-images.githubusercontent.com/3357792/68085443-76255c00-fe41-11e9-8f54-b050d5629af9.png)

**Comment by soredake** [comment]
@Zeioth please don't close valid issues, it's broken on all prefix, fresh or updated, vulkan is installed correctly, the problem is in the gallium nine option code.

**Comment by soredake** [comment]
@sonicpp can we discuss this on discord?

**Comment by sonicpp** [comment]
@soredake sure, I do not have access to internet connectiont atm, but tomorrow or friday should be ok (I started working on this alongside with resolving D9VK and Nine conflicts, but I didnt have much time...)

**Comment by Stebs** [comment]
Have the same problem, steps to reproduce:
Take a Wine-Prefix for any DX11 Game working flawlessly with enabled esync option.
Now enable the Gallium Nine option. (Game does not start).
Now disable the Gallium Nine option. Game does still not start.
Unless you disable also Esync option (or delete d3d9-nine.dll).
... So fooling around with the Gallium Nine option can prevent further usage of Esync option.
And as said above, it seems, upon lutris-wine update (and prefix update), d3d9-nine.dll is also sometimes added?
Isn't it possible to simply delete d3d9-nine.dll upon disabling Gallium Nine option?

**Comment by sonicpp** [comment]
I am not able to reproduce it even with Esync enabled. Strange. But it may be similar issue as #2526. I have tried several Wine versions, tried to switching on/off DXVK/Nine/Esync with no effect...

**Comment by soredake** [comment]
I moved to ubuntu, there is no wine-nine distro package, so cannot test.
